src/vsr/replica.zig

```diff
@@ -5807,8 +5940,64 @@ pub fn ReplicaType(
             const status_before = self.status;
             self.status = .view_change;
 
+            if (status_before == .normal and self.view_headers.array.get(0).op > self.op) {
+                // Transition from normal status, but the SV headers were part of the next wrap,
+                // so we didn't install them to our journal, and we didn't catch up.
+                // We will reuse the SV headers as our DVC headers to ensure that participating in
+                // another view-change won't allow the op to backtrack.
+                assert(self.log_view == self.view);
+
+                log.debug("{}: transition_to_view_change_status: " ++
+                    "(op_head={} view_headers_head={})", .{
+                    self.replica,
+                    self.op,
+                    self.view_headers.array.get(0).op,
+                });
+            } else if (status_before == .normal or
+                (status_before == .recovering and self.log_view == self.view) or
+                (status_before == .view_change and self.log_view == self.view))
+            {
+                // Either:
+                // - Transition from normal status.
+                // - Recovering from normal status.
+                // - Retired primary that didn't finish repair.
+                assert(self.log_view == self.view);
+
+                self.view_headers.array.len = 0;
+
+                const commit_max =
+                    std.math.max(self.commit_max, self.op -| constants.pipeline_prepare_queue_max);
```

Comment:
This feels weird a bit, can we just makit it an invariant that op can't be more thatn pipeline ahead of the commit_max?